src/geometry.F90

```diff
@@ -1483,11 +895,12 @@ subroutine neighbor_lists()
     do i = 1, n_cells
       c => cells(i)
 
-      ! loop over each surface specification
-      do j = 1, c % n_surfaces
-        i_surface = c % surfaces(j)
+      ! loop over each region specification
+      do j = 1, size(c%region)
+        i_surface = c % region(j)
         positive = (i_surface > 0)
         i_surface = abs(i_surface)
+        if (i_surface >= OP_UNION) cycle
```

Comment:
Add a comment to explain this.  Could be a short in-line comment like "This is an operator, not a region."